### PR TITLE
test/rgw/notifications: skip reload tests when there is no realm

### PR DESCRIPTION
**Pull request by yuvalif**
this does not impact teuthology, only local runs.
in teuthology all bucket notification tests are defined with a realm

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by yuvalif** [comment]
reload and migration tests are passing: 
https://pulpito.ceph.com/yuvalif-2024-05-23_05:51:29-rgw:notifications-wip-yuval-kafka-cleanup-distro-default-smithi/
failure in `test_notification_push_http` is expected: https://tracker.ceph.com/issues/66033

File: src/test/rgw/bucket_notification/test_bn.py [hunk 1]
```diff
@@ -4536,6 +4536,9 @@ def test_ps_s3_notification_push_kafka_security_sasl_scram():
 @attr('http_test')
 def test_persistent_ps_s3_reload():
     """ do a realm reload while we send notifications """
```
Comment (AliMasarweh) [inline src/test/rgw/bucket_notification/test_bn.py:4538]: this test is supposed to reload the realm while we send notifications,
however I see we only enable v2 notifications without disabling them at the start of the test
https://github.com/ceph/ceph/pull/57639/files#diff-83b1d2148e5a0483255b9af5f707fec576625a16c04aca191578d555ca336443R4617
but we can fix it in a different PR

**Review by AliMasarweh — APPROVED** [review]
LGTM

File: src/test/rgw/bucket_notification/test_bn.py (hunk 1 above)
Comment (yuvalif) [reply src/test/rgw/bucket_notification/test_bn.py:4538]: can you please open a tracker (we should backport to squid)  for this?

**Pull request merged**